src/main/java/ducky/util/Parser.java

```diff
@@ -0,0 +1,168 @@
+import ducky.command.ListCommand;
+import ducky.command.UpdateTaskCompletionCommand;
+import ducky.task.DeadlineTask;
+import ducky.task.EventTask;
+import ducky.task.Task;
+import ducky.task.TaskType;
+import ducky.task.TodoTask;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a Parser used for parsing commands and dates.
+ */
+public class Parser {
+
+    /**
+     * Constructs a Parser instance.
+     */
+    public Parser() {}
+
+    /**
+     * Parses the specified input and returns its Command representation, if applicable.
+     * @param cmd Command to be parsed.
+     * @return Command representation of specified input.
+     * @throws DuckyInvalidCommandException If the command does not exist.
+     * @throws DuckyInvalidCommandFormatException If the command is not in the right format.
+     */
+    public static Command parse(String cmd) throws DuckyInvalidCommandException, DuckyInvalidCommandFormatException {
+        String[] parts = cmd.split(" ", 2);
+        String commandType = parts[0].toLowerCase();
+        String argumentString = (parts.length > 1) ? parts[1].trim() : "";
+
+        switch (commandType) {
+        case "bye":
+            return new ExitCommand();
+        case "list":
+            return new ListCommand();
+        case "find":
+            return new FindTaskCommand(argumentString);
+        case "mark":
+            int markInputIndex;
+            try {
+                markInputIndex = Integer.parseInt(argumentString);
+            } catch (NumberFormatException e) {
+                throw new DuckyInvalidCommandFormatException("Did you enter a valid number?");
+            }
+            return new UpdateTaskCompletionCommand(markInputIndex, true);
+        case "unmark":
+            int unmarkInputIndex;
+            try {
+                unmarkInputIndex = Integer.parseInt(argumentString);
+            } catch (NumberFormatException e) {
+                throw new DuckyInvalidCommandFormatException("Did you enter a valid number?");
+            }
+            return new UpdateTaskCompletionCommand(unmarkInputIndex, false);
+        case "delete":
+            return new DeleteCommand(Integer.parseInt(argumentString));
+        case "todo":
+            // If description argument is empty, throw exception
+            if (argumentString.isEmpty()) {
+                throw new DuckyInvalidCommandFormatException(
+                        "A description is required for creating a to-do."
+                );
+            }
+
+            return new AddTaskCommand(TaskType.TODO, argumentString);
+        case "deadline":
+            String[] deadlineParts = argumentString.split("/by", 2);
+            // Check if there are 2 arguments
+            if (deadlineParts.length < 2) {
+                throw new DuckyInvalidCommandFormatException(
+                        "A description and deadline (in yyyy-mm-dd format) " +
```

Comment:
Good indentation for wrapped lines. But better to break before an operator (eg "+").